tests/src/core/testqgsdatasourceuri.cpp

```diff
@@ -492,6 +492,11 @@ void TestQgsDataSourceUri::checkAuthParams()
   QCOMPARE( uri4.param( QStringLiteral( "password" ) ), QStringLiteral( "😁😂😍" ) );
   QCOMPARE( uri4.password(), QStringLiteral( "😁😂😍" ) );
 
+  // issue GH #53654
+  QgsDataSourceUri uri5;
+  uri5.setEncodedUri( QStringLiteral( "zmax=14&zmin=0&styleUrl=http://localhost:8000/&f=application%2Fvnd.geoserver.mbstyle%2Bjson" ) );
+  QCOMPARE( uri5.param( QStringLiteral( "f" ) ), QStringLiteral( "application%2Fvnd.geoserver.mbstyle%2Bjson" ) );
+
```

Comment:
Could use also add a test where an explicit "+" character is used in the value for a parameter?

Reply:
Good idea, done in https://github.com/qgis/QGIS/pull/53701/commits/4bb4979a1021509c63d63d3acb3464e17f609157.

Reply:
Sorry - one last test: can you create a QgsDataSourceUri, and then set two parameters for it using setParam, one with a %2B literal in the value and another with a + literal, and see if the values survive the round trip through encodedUri/setEncodedUri losslessly?

Reply:
:+1: , they do survive, test added in https://github.com/qgis/QGIS/pull/53701/commits/8aa299c569acebb5a4c62f2792f672ff57544d79.